From: Matthew Gamble <git@matthewgamble.net>
Date: Sat, 19 Mar 2016 15:17:33 +1100
Subject: [PATCH] Remove hard dependency on pip

---
There's some magic in this package that attempts to automatically install
packages from PyPI if they're not already installed. This doesn't work
unless you're running your code with root privileges, so there's no point
in having it here. These packages should instead be installed using
your system's package manager.

Also, we make the assumption that importlib will always exist because
this package is targeting python3 on Arch Linux.

diff -u -r luckydonaldUtils/dependencies.py luckydonaldUtils/dependencies.py
--- luckydonaldUtils/dependencies.py	2016-03-02 10:59:03.000000000 +1100
+++ luckydonaldUtils/dependencies.py	2016-03-19 15:14:54.191287535 +1100
@@ -3,17 +3,7 @@
 
 logger = logging.getLogger(__name__)
 
-import pip
-
-try:
-    import importlib
-except ImportError:
-    # pip install importlib
-    pip.main(["install", "importlib"])
-    import importlib
-
-
-# end try
+import importlib
 
 
 def import_or_install(package_name, pip_name=None):
diff -u -r setup.py setup.py
--- setup.py	2016-03-03 08:44:22.000000000 +1100
+++ setup.py	2016-03-19 15:13:50.879930426 +1100
@@ -12,12 +12,6 @@
 
 main_package = 'luckydonaldUtils'
 
-install_requires = ["pip", "setuptools"]  # both should be installed by default.
-try:
-    import importlib
-except ImportError:
-    install_requires.append('importlib')
-
 setup(
     name="luckydonald-utils",
     packages=find_submodules(main_package),
@@ -28,7 +22,6 @@
     license="GPLv3+",
     keywords="luckydonald utils utilities utility python dependencies dependency download progress bar encoding files interactions json update store text xml time network logger color images webserver django CSRF headers eastereggs",
     url="https://github.com/luckydonald/python-utils",
-    install_requires=install_requires,
     long_description=long_description,
     test_suite = 'nose.collector',
     classifiers=[
